moto/elbv2/models.py

```diff
@@ -262,50 +256,103 @@ def create_from_cloudformation_json(
-                            + "._"
-                            + camelcase_to_underscores(redirect_config_key)
-                        ] = redirect_config_value
-                    default_actions.append(redirect_action)
-                else:
-                    raise InvalidActionTypeError(action_type, i + 1)
-        else:
-            default_actions = None
+        default_actions = []
+        for i, action in enumerate(properties["DefaultActions"]):
+            action_type = action["Type"]
+            if action_type == "forward":
+                default_actions.append(
+                    {"type": action_type, "target_group_arn": action["TargetGroupArn"],}
+                )
+            elif action_type in [
+                "redirect",
+                "authenticate-cognito",
+                "fixed-response",
+            ]:
+                redirect_action = {"type": action_type}
+                key = (
+                    underscores_to_camelcase(action_type.capitalize().replace("-", "_"))
+                    + "Config"
+                )
+                for redirect_config_key, redirect_config_value in action[key].items():
+                    # need to match the output of _get_list_prefix
+                    redirect_action[
+                        camelcase_to_underscores(key)
+                        + "._"
+                        + camelcase_to_underscores(redirect_config_key)
+                    ] = redirect_config_value
+                default_actions.append(redirect_action)
+            else:
+                raise InvalidActionTypeError(action_type, i + 1)
 
         listener = elbv2_backend.create_listener(
             load_balancer_arn, protocol, port, ssl_policy, certificates, default_actions
         )
         return listener
 
 
+class FakeListenerRule(CloudFormationModel):
+    def __init__(
+        self, listener_arn, arn, conditions, priority, actions,
+    ):
+        self.listener_arn = listener_arn
+        self.arn = arn
+        self.conditions = conditions
+        self.actions = actions
+        self.priority = priority
+
+    @property
+    def physical_resource_id(self):
+        return self.arn
+
+    @staticmethod
+    def cloudformation_type():
+        # https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-elasticloadbalancingv2-listenerrule.html
+        return "AWS::ElasticLoadBalancingV2::ListenerRule"
+
+    @classmethod
+    def create_from_cloudformation_json(
+        cls, resource_name, cloudformation_json, region_name
+    ):
+        properties = cloudformation_json["Properties"]
+        elbv2_backend = elbv2_backends[region_name]
+        listener_arn = properties.get("ListenerArn")
+        priority = properties.get("Priority")
+        conditions = properties.get("Conditions")
+        # transform Actions to confirm with the rest of the code and XML templates
+        default_actions = []
+        for i, action in enumerate(properties["Actions"]):
```

Comment:
The logic here seems to be the same as when creating a AWS::ElasticLoadBalancingV2::Listener, the only difference is whether it's called `Actions` or `DefaultActions`.
Can we extract this while loop in a common method, and reuse it for both?

Reply:
I did try and then reverted it: https://github.com/spulec/moto/pull/3969/commits/704a5ab890836d6588dd652d58475c2d0d08bd6f
I do not remember why. Will try it again

Reply:
Oh yes. The Action works a little differently in ListernRule. My intention was to fix that in the next PR.
Here is how: There is a ForwardConfig option in the ListenerRule that in-turn defines the TargetGroups and the TargetGroupStickinessConfig. That needs to be added in the check. So really this will need to change when adding proper validations.
What do you think?

Reply:
Example:
```
LoadBalancerListenerRule:
    Type: AWS::ElasticLoadBalancingV2::ListenerRule
    Properties:
      Actions:
        - Type: forward
          ForwardConfig:
            TargetGroups:
              - TargetGroupArn:
                  {{target_group.arn}}
                Weight: {{target_group.weight}}
            TargetGroupStickinessConfig:
              DurationSeconds: 300
              Enabled:
                Fn::If: [ IsStickinessEnabled, true, false ]
      Conditions:
          - Field: path-pattern
            Values:
              - '/*'
      ListenerArn: 
        Ref: LoadBalancerListener
      Priority: 1
```
This, right now, fails with the error:
KeyError: 'TargetGroupArn'

Reply:
I understand ( ..I think). If future behaviour will create a difference between them, there's no point in making it DRY now.